lib/web/apiserver.go

```diff
@@ -1659,25 +1659,28 @@ type renewSessionRequest struct {
 	AccessRequestID string `json:"requestId"`
 	// Switchback indicates switching back to default roles when creating new session.
 	Switchback bool `json:"switchback"`
+	// UpdateUserTraits is a flag to indicate if user traits need to be updated.
+	UpdateUserTraits bool `json:"updateUserTraits"`
 }
 
 // renewSession updates this existing session with a new session.
 //
 // Depending on request fields sent in for extension, the new session creation can vary depending on:
-//   - requestId (opt): appends roles approved from access request to currently assigned roles or,
-//   - switchback (opt): roles stacked with assuming approved access requests, will revert to user's default roles
+//   - AccessRequestID (opt): appends roles approved from access request to currently assigned roles or,
+//   - Switchback (opt): roles stacked with assuming approved access requests, will revert to user's default roles
+//   - UpdateLoginTraits (opt): similar to default but updates the login traits by retrieving it from the backend
 //   - default (none set): create new session with currently assigned roles
 func (h *Handler) renewSession(w http.ResponseWriter, r *http.Request, params httprouter.Params, ctx *SessionContext) (interface{}, error) {
 	req := renewSessionRequest{}
 	if err := httplib.ReadJSON(r, &req); err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	if req.AccessRequestID != "" && req.Switchback {
-		return nil, trace.BadParameter("Failed to renew session: fields 'AccessRequestID' and 'Switchback' cannot be both set")
+	if req.AccessRequestID != "" && req.Switchback || req.AccessRequestID != "" && req.UpdateUserTraits || req.Switchback && req.UpdateUserTraits {
```

Comment:
This will grow exponentially :sweat_smile: 
I guess we didn't pick the best structure to represent that a single action can be performed.
Not sure how we can improve now.

On another note, we could add a `CheckAndSetDefaults` method to the `renewSessionRequest` strcut where we would do this check.

Reply:
the `ExtendWebSession` endpoint will eventually need to get converted to grpc, so I didn't want to worry about it (it calls on the deprecated http auth endpoint). i was going to convert it and use `oneof` in the proto message, but when i started doing it, it wasn't a straightforward conversion, and I didn't want to take a lot of time refactoring/testing at this time when discovery is due soon